arm9/source/game/cia.c

```diff
@@ -59,21 +60,49 @@ u32 BuildCiaCert(u8* ciacert) {
         0x18, 0x83, 0xAF, 0xE0, 0xF4, 0xE5, 0x62, 0xBA, 0x69, 0xEE, 0x72, 0x2A, 0xC2, 0x4E, 0x95, 0xB3
     };
 
-    // open certs.db file on SysNAND
-    FIL db;
-    UINT bytes_read;
-    if (f_open(&db, "1:/dbs/certs.db", FA_READ | FA_OPEN_EXISTING) != FR_OK)
+    const char* issuer_ca = !IS_DEVKIT ? "Root-CA00000003" : "Root-CA00000004";
```

Comment:
This applies to `ticket.c` and `tmd.c` as well: I see a lot of repeated code here. Wouldn't it be possible to do all the repeating stuff in a function in certs.c? You shouldn't even have a lot of parameters here.

Reply:
It can be, but given that `cia.c` also loads the same way, but 3 certs, I could make a function that takes a variable argument list (`...`), or a list of certificate names in a `const char**` list terminated by a NULL or a count argument, that then generate a raw certificate bundle.

Reply:
Commit 5b27b5a, asides a bit of refactoring, implements a cleaner call, and more optimized, as there's no repeated calls of `LoadCertFromCertDb` as effect.